cdap-hbase-compat-base/src/main/java/co/cask/cdap/data2/util/hbase/HBaseTableUtil.java

```diff
@@ -239,8 +245,8 @@ public void createTableIfNotExists(HBaseAdmin admin, TableId tableId, HTableDesc
     setDefaultConfiguration(tableDescriptor, admin.getConfiguration());
     tableDescriptor = setVersion(tableDescriptor);
     tableDescriptor = setTablePrefix(tableDescriptor);
-    DefaultHBaseDDLExecutor executor = (DefaultHBaseDDLExecutor) getHBaseDDLExecutor(admin.getConfiguration());
-    TableDescriptor descriptor = executor.getTableDescriptor(tableDescriptor);
+    HBaseDDLExecutor executor = new HBaseDDLExecutorFactory(cConf, admin.getConfiguration()).get();
+    TableDescriptor descriptor = ((DefaultHBaseDDLExecutor) executor).getTableDescriptor(tableDescriptor);
```

Comment:
yes this will be done in separate PR. where createTable method will directly accept `TableDescriptor` then this conversion is not needed.